cmake/Boost.cmake

```diff
@@ -1,34 +1,71 @@
 message("* boost")
-set (BOOST_SOURCE_DIR "${POL_EXT_LIB_DIR}/boost_1_67_0")
+set (BOOST_SOURCE_DIR "${CMAKE_CURRENT_LIST_DIR}/../lib/boost_1_74_0")
+set (BOOST_STAGE_LIB_DIR "${BOOST_SOURCE_DIR}/stage/lib")
 
-if(NOT EXISTS "${BOOST_SOURCE_DIR}/boost")
-  # build disabled no need currenty
-  ExternalProject_Add(boost
-	URL "${BOOST_SOURCE_DIR}/../boost_1_67_0.tar.bz2"
-    SOURCE_DIR "${BOOST_SOURCE_DIR}" 
-#	CONFIGURE_COMMAND ./bootstrap.sh
-#       --prefix=build
-	INSTALL_COMMAND ""
-    CONFIGURE_COMMAND ""
-    BUILD_COMMAND ""
-#	BUILD_COMMAND ./bjam
-#       toolset=gcc
-#       link=shared
-#       runtime-link=shared
-#       address-model=${ARCH_BITS}
-#       variant=release
-#       --abbreviate-paths
-#       --stagedir=lib
-#       --build-dir=build
-#       --with-regex
-#       stage
-	LOG_DOWNLOAD 1
-    BUILD_IN_SOURCE 1
-  )
-  set_target_properties (boost PROPERTIES FOLDER 3rdParty)
+if (clang)
+  set (BOOST_TOOLSET "clang")
+elseif (gcc)
+  set (BOOST_TOOLSET "gcc")
+elseif (msvc)
+  if (MSVC_TOOLSET_VERSION EQUAL 140)
+    set (BOOST_TOOLSET "msvc") # VS 2015 (14.0)
+  elseif(MSVC_TOOLSET_VERSION GREATER_EQUAL 141)
+    set(BOOST_TOOLSET "msvc-14.1") # VS 2017 (15.0) and VS 2019 (16.0)
+  endif()
+endif()
+
+if (NOT DEFINED BOOST_TOOLSET)
+  message(FATAL_ERROR "Unknown boost toolset to build")
+endif()
+
+if (${windows})
+  set (BOOST_CONFIGURE_COMMAND "bootstrap.bat")
+  set (BOOST_BUILD_COMMAND "b2.exe")
+  set (BOOST_REGEX_LIB "${BOOST_STAGE_LIB_DIR}/libboost_regex.lib" )
+  set (BOOST_SYSTEM_LIB "${BOOST_STAGE_LIB_DIR}/libboost_system.lib" )
+  set (BOOST_THREAD_LIB "${BOOST_STAGE_LIB_DIR}/libboost_thread.lib" )
 else()
-  message("Boost already extracted")
+  set (BOOST_CONFIGURE_COMMAND "./bootstrap.sh")
+  set (BOOST_BUILD_COMMAND "./b2")
+  set (BOOST_REGEX_LIB "${BOOST_STAGE_LIB_DIR}/libboost_regex.a" )
+  set (BOOST_SYSTEM_LIB "${BOOST_STAGE_LIB_DIR}/libboost_system.a" )
+  set (BOOST_THREAD_LIB "${BOOST_STAGE_LIB_DIR}/libboost_thread.a" )
 endif()
 
-set (BOOST_LIBRARY_DIR "${BOOST_SOURCE_DIR}/lib/lib")
+if(NOT EXISTS ${BOOST_REGEX_LIB} OR NOT EXISTS ${BOOST_SYSTEM_LIB} OR NOT EXISTS ${BOOST_THREAD_LIB})
+  message("  - will build in ${BOOST_SOURCE_DIR} toolset=${BOOST_TOOLSET}")
+  ExternalProject_Add(boost
```

Comment:
The boost folder should be cached in our CI like the other bigger 3rd party libs.

Reply:
Sounds good.  How?

Reply:
In the yml files in .GitHub/workflow is something like:
- name: Cache antlr
        uses: actions/cache@v1
        with:
          path: lib/antlr
          key: ${{ runner.os }}-${{ matrix.config.cxx-compiler }}-${{ matrix.config.compiler-version }}-antlr-buildtypefix

(And the same for curl)
Basically just copy it and adapt the name and the path. The key is something unique which (hopefully) triggers a rebuild when something changes which needs a rebuild. In this case I guess the os and compiler version is enough since the boost version is part of the directory name.